### PR TITLE
Airflow kubernetes cli command for generating pod templates

### PR DESCRIPTION
**Pull request by Javier162380**
Adding a new command to the Airflow CLI.
``` bash
airflow kubernetes pod-preview dag-id --sub-dir
```
The idea is to be able to generate pod templates from KubernetesPodOperator, so given a target dag it can generate all the k8s pod templates and printing in the CLI.

Example.

image
image

Just one consideration I am not connecting to any Kubernetes client so I am generating a default service account which means that you can not launch this pod in a Kubernetes cluster with a command like this.
```
kubectl apply -f file.yaml
```
Any feedback here is appreciated.

Also to be able to test this feature I need to add a new example-dag, 
``` bash
example_kubernetes_pod_operator.py
```
It is the same example given by Google in their composer tutorial (https://cloud.google.com/composer/docs/how-to/using/using-kubernetes-pod-operator), but adapting the imports to the new Kubernetes module import system.

---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In the case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backward incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by potiuk** [comment]
I think you will need to rebase it, we had some fixes in master!

**Comment by Javier162380** [comment]
> I think you will need to rebase it, we had some fixes in master!

ok this can explain why the travis test are failing?

**Comment by potiuk** [comment]
Some yes. Possibly not all.

**Comment by Javier162380** [comment]
> Some yes. Possibly not all.

ok I already rebase it let's hope this time tests pass

**Comment by Javier162380** [comment]
the integration tests are still failing any clue @potiuk ?

**Comment by mik-laj** [comment]
@Javier162380  There seems to be a serialization error not related to this patch. I have prepared a small change that bypasses this problem.
```
curl https://termbin.com/9q5d | git am
```
```diff
From a9c150fe70b57e013d4bb106638caaf92a2e214b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kamil=20Bregu=C5=82a?= <kamil.bregula@polidea.com>
Date: Mon, 20 Apr 2020 18:55:28 +0200
Subject: Fix serialization

---
 airflow/example_dags/example_kubernetes_pod_operator.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/airflow/example_dags/example_kubernetes_pod_operator.py b/airflow/example_dags/example_kubernetes_pod_operator.py
index f3b18cc13..74a7c4cab 100644
--- a/airflow/example_dags/example_kubernetes_pod_operator.py
+++ b/airflow/example_dags/example_kubernetes_pod_operator.py
@@ -20,8 +20,9 @@ import datetime
 from airflow import models
 from airflow.kubernetes import secret
 from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
+from airflow.utils.dates import days_ago
 
-YESTERDAY = datetime.datetime.now() - datetime.timedelta(days=1)
+YESTERDAY = days_ago(1)
 secret_env = [secret.Secret(
     deploy_type='env',
     deploy_target='SQL_CONN',
-- 
2.20.1

```

**Comment by Javier162380** [comment]
> @Javier162380 There seems to be a serialization error not related to this patch. I have prepared a small change that bypasses this problem.
> 
> ```
> curl https://termbin.com/9q5d | git am
> ```
> 
> ```diff
> From a9c150fe70b57e013d4bb106638caaf92a2e214b Mon Sep 17 00:00:00 2001
> From: =?UTF-8?q?Kamil=20Bregu=C5=82a?= <kamil.bregula@polidea.com>
> Date: Mon, 20 Apr 2020 18:55:28 +0200
> Subject: Fix serialization
> 
> ---
>  airflow/example_dags/example_kubernetes_pod_operator.py | 3 ++-
>  1 file changed, 2 insertions(+), 1 deletion(-)
> 
> diff --git a/airflow/example_dags/example_kubernetes_pod_operator.py b/airflow/example_dags/example_kubernetes_pod_operator.py
> index f3b18cc13..74a7c4cab 100644
> --- a/airflow/example_dags/example_kubernetes_pod_operator.py
> +++ b/airflow/example_dags/example_kubernetes_pod_operator.py
> @@ -20,8 +20,9 @@ import datetime
>  from airflow import models
>  from airflow.kubernetes import secret
>  from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
> +from airflow.utils.dates import days_ago
>  
> -YESTERDAY = datetime.datetime.now() - datetime.timedelta(days=1)
> +YESTERDAY = days_ago(1)
>  secret_env = [secret.Secret(
>      deploy_type='env',
>      deploy_target='SQL_CONN',
> -- 
> 2.20.1
> ```

thanka! I don't was going wrong, to be honest

**Comment by turbaszek** [comment]
@dimberman is this something that may help us for migration to Airflow 2.0?

**Comment by dimberman** [comment]
@turbaszek @Javier162380 I'm a bit confused, how is this different from this command?

```
(airflow) new-features  (scheduler-ha-helm-master)  (⎈ |kind-airflow-python-3.6-v1.18.6|default) $airflow kubernetes generate-dag-yaml -h
usage: airflow kubernetes generate-dag-yaml [-h] [-o OUTPUT_PATH] [-S SUBDIR]
                                            dag_id execution_date

Generate YAML files for all tasks in DAG. Useful for debugging tasks without launching into a cluster

positional arguments:
  dag_id                The id of the dag
  execution_date        The execution date of the DAG

optional arguments:
  -h, --help            show this help message and exit
  -o OUTPUT_PATH, --output-path OUTPUT_PATH
                        The output for generated yaml files
  -S SUBDIR, --subdir SUBDIR
                        File location or directory from which to look for the dag. Defaults to '[AIRFLOW_HOME]/dags' where [AIRFLOW_HOME] is the value you set for 'AIRFLOW_HOME' config you set in 'airflow.cfg'
```

File: airflow/cli/cli_parser.py [hunk 1]
```diff
@@ -1026,6 +1026,14 @@ class GroupCommand(NamedTuple):
         args=(ARG_ROLES,),
     ),
 )
+KUBERNETES_COMMANDS = (
+    ActionCommand(
+        name='pod-preview',
+        help='Generate a k8s template',
```
Comment (dimberman) [inline airflow/cli/cli_parser.py:1032]:
```suggestion
        help='Generate a k8s template for KubernetesPodOperator tasks',
```

File: airflow/cli/commands/kubernetes_command.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    # kubernetes classes args to arguments and env_vars to envs.
+    if 'args' not in pod_args:
+        pod_args['args'] = kubernetes_tasks_args.get('arguments', None)
+    if 'envs' not in pod_args:
+        pod_args['envs'] = kubernetes_tasks_args.get('env_vars', None)
+    if 'extract_xcom' not in pod_args:
+        pod_args['extract_xcom'] = kubernetes_tasks_args.get('do_xcom_push', None)
+
+    pod = pod_generator.PodGenerator(**(pod_args)).gen_pod()
+    pod = append_to_pod(pod,
```
Comment (dimberman) [inline airflow/cli/commands/kubernetes_command.py:63]: Why not just use this existing function in the KubernetesPodOperator?

https://github.com/apache/airflow/blob/master/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L354

**Review by dimberman — CHANGES_REQUESTED** [review]

**Comment by dimberman** [comment]
@Javier162380 do you want to finish this? If not I can pick this up :)

**Comment by Javier162380** [comment]
> @Javier162380 do you want to finish this? If not I can pick this up :)

I can help you with! Need to reclone the repo it is quite oldie this PR

**Comment by dimberman** [comment]
Hi @Javier162380 are you interested in continuing with this task? Glad to help :).

**Comment by ashb** [comment]
@dimberman Didn't you add something like this already?

**Comment by mik-laj** [comment]
We already have ``airflow kubernetes airflow kubernetes generate-dag-yaml`` command. Closing...

**Pull request closed without merging**